rskj-core/src/main/java/org/ethereum/net/client/ConfigCapabilitiesImpl.java

```diff
@@ -29,26 +29,33 @@
 import java.util.TreeSet;
 
 import static org.ethereum.net.client.Capability.RSK;
+import static org.ethereum.net.client.Capability.SNAP;
 import static org.ethereum.net.eth.EthVersion.fromCode;
 
 /**
  * Created by Anton Nashatyrev on 13.10.2015.
  */
 public class ConfigCapabilitiesImpl implements ConfigCapabilities{
 
-    private final SystemProperties config;
+    private final RskSystemProperties config;
 
     private SortedSet<Capability> allCaps = new TreeSet<>();
 
-    public ConfigCapabilitiesImpl(SystemProperties config) {
+    public ConfigCapabilitiesImpl(RskSystemProperties config) {
         if (config.syncVersion() != null) {
             EthVersion eth = fromCode(config.syncVersion());
             if (eth != null) {
                 allCaps.add(new Capability(RSK, eth.getCode()));
+                if (config.isServerSnapshotSyncEnabled()){
+                    allCaps.add(new Capability(SNAP, eth.getCode()));
+                }
             }
         } else {
             for (EthVersion v : EthVersion.supported()) {
                 allCaps.add(new Capability(RSK, v.getCode()));
+                if (config.isServerSnapshotSyncEnabled()){
+                    allCaps.add(new Capability(SNAP, v.getCode()));
```

Comment:
I guess it should be version `1` for `snap`, as `snap` and `rsk` protocol versions are kinda independent

Also I'd suggest to extract adding `snap` capability out of this `if-else`, eg:
```
        if (config.isServerSnapshotSyncEnabled() && allCaps.stream().anyMatch(Capability::isRSK)) {
            allCaps.add(new Capability(SNAP, (byte) 1));
        }
```

Reply:
addressed, thanks for the suggestion! I've implemented it so that the SNAP capability now uses version` (byte) 1`